tests/e2e_decorrelation/test_error_cases.py

```
@@ -0,0 +1,715 @@
+from .test_utils import (
+    assert_plan_structure,
+    assert_result_count,
+    assert_result_contains_ids,
+    execute_and_fetch_all
+)
+
+
+class TestCardinalityViolations:
+    """Test cardinality enforcement errors."""
+
+    def test_scalar_subquery_multiple_rows_error(self, catalog, setup_test_data):
```

Ruff (0.14.5)

Unused method argument: `setup_test_data`

(ARG002)